mm2src/coins/nft.rs

```diff
@@ -78,24 +81,60 @@ pub type WithdrawNftResult = Result<TransactionNftDetails, MmError<WithdrawError
 ///
 /// Returns `GetNftInfoError` variants for issues like invalid requests, transport failures,
 /// database errors, and spam protection errors.
-pub async fn get_nft_list(ctx: MmArc, req: NftListReq) -> MmResult<NftList, GetNftInfoError> {
-    let nft_ctx = NftCtx::from_ctx(&ctx).map_to_mm(GetNftInfoError::Internal)?;
+pub async fn get_nft_list(ctx: MmArc, req: NftListReq) -> MmResult<Vec<NftLists>, GetNftInfoError> {
+    let db_ids = find_unique_nft_account_ids(&ctx, req.chains.clone())
+        .await
+        .map_to_mm(GetNftInfoError::Internal)?;
+
+    let get_nfts =
+        |id: String, chains: Vec<Chain>| -> Pin<Box<dyn Future<Output = MmResult<NftLists, GetNftInfoError>> + Send>> {
+            let ctx_clone = ctx.clone();
+            let req = req.clone();
+
+            let res = async move {
+                let nft_ctx = NftCtx::from_ctx(&ctx_clone).map_to_mm(GetNftInfoError::Internal)?;
+
+                let chains = req
+                    .chains
+                    .clone()
+                    .into_iter()
+                    .filter(|c| chains.contains(c))
+                    .collect::<Vec<_>>();
+                let storage = nft_ctx.lock_db(Some(&id)).await?;
```

Comment:
> Also I guess we wanna take the approach of using one DB for everything

I asked the same question.
The answer was that we need this multiple db approach for external wallets when we can't get the master key but only account keys

Reply:
> > Also I guess we wanna take the approach of using one DB for everything
> 
> I asked the same question. The answer was that we need this multiple db approach for external wallets when we can't get the master key but only account keys

we need this to store the account's swaps data.

Reply:
>> there is only one async DB under the hood

There isn't only one async DB under the hood, more can be initialized with external wallet. `lock_db` is calling https://github.com/KomodoPlatform/komodo-defi-framework/blob/8f19481a222e96a3073c059377023a7ffc72888e/mm2src/mm2_core/src/sql_connection_pool.rs#L241 internally to open new connection or get if it's already opened.. So I don't quite get your comment very clear

Reply:
Sorry I should have linked this:
https://github.com/KomodoPlatform/komodo-defi-framework/blob/8f19481a222e96a3073c059377023a7ffc72888e/mm2src/mm2_core/src/sql_connection_pool.rs#L193-L201

So I see here that we open the same `pool.sqlite_file_path` for every `db_id` no matter what the `db_id` is. I thought this is intentional, but upon a second look it looks like it's not.

Reply:
> Also I guess we wanna take the approach of using one DB for everything.

I meant here a single DB arch per key (so per mm2 master key (i.e mm2 instance) or external wallet key (pubkey since we don't get to see the private)).
I think right now we do one DB per account for the HD case, but I could be corrected on that.

Reply:
> Sorry I should have linked this:
> 
> https://github.com/KomodoPlatform/komodo-defi-framework/blob/8f19481a222e96a3073c059377023a7ffc72888e/mm2src/mm2_core/src/sql_connection_pool.rs#L193-L201
> 
> So I see here that we open the same `pool.sqlite_file_path` for every `db_id` no matter what the `db_id` is. I thought this is intentional, but upon a second look it looks like it's not.

I added additional checks for this before opening a new db connection for a `db_id`

Reply:
> I meant here a single DB arch per key (so per mm2 master key (i.e mm2 instance) or external wallet key (pubkey since we don't get to see the private))

I believe using the `rmd160` hash of the pubkey is enough for external wallets so we don't necessarily need the master before we can generate a `db_id` for them

Reply:
> Also I guess we wanna take the approach of using one DB for everything. So not create multiple async DBs. <- cc/ @shamardy

Sorry for the late reply @mariocynicys , one DB for everything is not a bad idea, but we can't foresee what problems it will create in the future. For instance, when implementing cloud data backup that is encrypted using the seed, we will have to go through this one folder/db and only choose the data for this seed which will be a hard implementation for file system db. And I think I discussed this with you before, backward compatibility for iguana seeds with one DB will not work. I don't like the current complicated architecture in this PR and I will try to make it better every review iteration but I don't see a way around having multiple dbs unfortunately :(

Reply:
> The answer was that we need this multiple db approach for external wallets when we can't get the master key but only account keys

We actually can if we have a column for account keys used or similar approaches, but we will use complex select statements, etc..

Reply:
P.S. current architecture in this PR gives the most flexibility, with it we can even implement enabling different coins with different mm2 mnemonics in the future if needed :)